### DISCUSSION TITLE
buildSrc dependecies in specific format are not detected

### DISCUSSION THREAD

**Discussion by jaroslavhavelik** [renovatebot/renovate#24246] [Request Help] [answered]
### Tell us more.

I have multi module setup with dependencies managed in following manner:

```
object AndroidCore {
    private const val APP_COMPAT = "1.6.1"
    private const val CORE_KTX = "1.9.0"
    private const val FRAGMENT_KTX = "1.6.0"
    private const val LIFECYCLE_KTX = "2.6.1"
    private const val WEBKIT = "1.7.0"

    private val dependencies = listOf(
        "androidx.appcompat:appcompat:$APP_COMPAT",
        "androidx.core:core-ktx:$CORE_KTX",
        "androidx.fragment:fragment-ktx:$FRAGMENT_KTX",
        "androidx.lifecycle:lifecycle-livedata-ktx:$LIFECYCLE_KTX",
        "androidx.lifecycle:lifecycle-viewmodel-ktx:$LIFECYCLE_KTX",
        "androidx.webkit:webkit:$WEBKIT",
    )

    fun DependencyHandler.androidCore(configurationName: String = "implementation") =
        dependencies.forEach { add(configurationName, it) }
}
```

In each module extension function is used:

```
dependencies {
    androidCore()
}

```
But renovate doesn't detect these kinds of dependencies. Other dependencies that just refer to version constants are working but these are not. I suspect that it's not a problem of `buildSrc` itself, but this specific format.

Can anybody advice what I can do in order to renovate detect also these dependencies?

**Comment by viceice** [comment]
I don't think we can support that kind of custom code. You probably need to use regex manager to parse those deps manually.

@Churro @zharinov WDYT?

**Reply by zharinov** [reply]
We don't track function calls and any other computation constructs, but I remember we treated string literals like `"foo:bar:$BAZ"` as dependencies. Probably, we can track why this isn't the case here and fix it.

**Comment by Churro** [comment, marked as answer]
It should be feasible to iterate over values in calls to `listOf`. Support for this is currently missing but should be no big deal to add. Feel free to assign this to me.

**Reply by rarkins** [reply]
Thanks @Churro. Created and assigned #24279
